### PR TITLE
Re-implement template refactor

### PR DESCRIPTION
**Pull request by chcosta**
### To double check:

* [ ] The right tests are in and and the right validation has happened.  Guidance:  https://github.com/dotnet/arcade/tree/main/Documentation/Validation

This is the revert of [this](https://github.com/dotnet/arcade/commit/87b015b938e5400d6e57afd7650348c17a764b73) revert.

The first implementation [failed](https://dnceng.visualstudio.com/internal/_build/results?buildId=2428680&view=logs&j=29d4eac0-e772-56a4-ad0d-2d727d908ceb&t=4edcb01d-2da8-54c4-439e-584d3c1a5e90&s=5eb52e0a-bed2-568e-0700-89f43bce9ba8) because I didn't realize teams were making direct references to some of the post-build templates.

this change adds shims for those templates which were directly referenced and caused the publishing failure.

Validated the code path works with [this](https://dev.azure.com/dnceng/internal/_build/results?buildId=2429330&view=logs&j=226748d0-f812-5437-d3f0-2dd291f5666e&t=bad11196-972e-5d03-45a8-9db526506073) arcade job.  The job fails because i didn't configure the temporary channel i set up properly, but the desired code path is executed.

I'll do some more validation today / this weekend.  I have no intention of merging this on a Friday afternoon.

### CODE REVIEW

File: eng/common/core-templates/steps/component-governance.yml [hunk 1]
```diff
@@ -1,10 +1,11 @@
 parameters:
   disableComponentGovernance: false
   componentGovernanceIgnoreDirectories: ''
-
+  is1ESPipeline: false
+  
 steps:
 - ${{ if eq(parameters.disableComponentGovernance, 'true') }}:
-  - script: "echo ##vso[task.setvariable variable=skipComponentGovernanceDetection]true"
+  - script: echo "##vso[task.setvariable variable=skipComponentGovernanceDetection]true"
```
Comment (akoeplinger) [inline eng/common/core-templates/steps/component-governance.yml:8]: this is not correct, it reverts https://github.com/dotnet/arcade/pull/14687

File: eng/common/core-templates/job/publish-build-assets.yml [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+        targetType: inline
+        script: |
+          $symbolExclusionfile = "$(Build.SourcesDirectory)/eng/SymbolPublishingExclusionsFile.txt"
+          if(Test-Path -Path $symbolExclusionfile)
+          {
+            Write-Host "SymbolExclusionFile exists"
+            Write-Host "##vso[task.setvariable variable=SymbolExclusionFile]true"
+          }
+          else{
+           Write-Host "Symbols Exclusion file does not exists"
```
Comment (MattGal) [inline eng/common/core-templates/job/publish-build-assets.yml:135]: nit: agreement (a singular object does not exis**t.**)

**Review by missymessa — DISMISSED** [review]

**Review by AlitzelMendez — DISMISSED** [review]

**Comment by missymessa** [comment]
GOOD LUCK!

**Review by AlitzelMendez — APPROVED** [review]

**Review by epananth — DISMISSED** [review]

**Review by AlitzelMendez — APPROVED** [review]

**Comment by chcosta** [comment]
> looks like we'll need a third try

let's not limit ourselves.

jk, i understand why my validation failed, and i'm enlisting @epananth to perform validation as well.  sorry for the noise!

**Pull request merged**